test/unit/volume_manager_btrfs_test.py

```diff
@@ -13,6 +16,22 @@
 class TestVolumeManagerBtrfs(object):
     @patch('os.path.exists')
     def setup(self, mock_path):
+        self.context_manager_mock = mock.Mock()
+        self.file_mock = mock.Mock()
+        self.enter_mock = mock.Mock()
+        self.exit_mock = mock.Mock()
+        self.enter_mock.return_value = self.file_mock
+        setattr(self.context_manager_mock, '__enter__', self.enter_mock)
+        setattr(self.context_manager_mock, '__exit__', self.exit_mock)
+        self.xml_info = dedent('''
+        <?xml version="1.0" ?>
+        <snapshot>
+            <type>single</type>
+            <num>1</num>
+            <description>first root filesystem</description>
+            <date>2016-01-01 00:00:00</date>
+        </snapshot>
+        ''').strip() + os.linesep
```

Comment:
Maybe it will work for you too: in Salt unit tests we are trying remove these kind of data outside into a text files and then load on demand.